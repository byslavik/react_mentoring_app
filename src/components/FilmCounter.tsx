import * as React from 'react';

import {FilmCounterProps, FilmCounterState} from '../models/film.model';

export class FilmCounter extends React.Component<FilmCounterProps, FilmCounterState> {

    render() {
      let count = this.props.count;

      let text = count == 1 ? 'movie': 'movies';
      
      return (<span>{count} {text} found</span>)
    }
}
