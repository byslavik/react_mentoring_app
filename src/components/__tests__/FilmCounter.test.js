import * as React from 'react';
import {FilmCounter} from '../FilmCounter';
import * as renderer from 'react-test-renderer';

describe('FilmCounter component', ()=>{
  it('FilmCounter component shows correctly with 1 movie', () => {
    const tree = renderer.create(
      <FilmCounter count={1}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('FilmCounter component shows correctly with many movies', () => {
    const tree = renderer.create(
      <FilmCounter count={2}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});