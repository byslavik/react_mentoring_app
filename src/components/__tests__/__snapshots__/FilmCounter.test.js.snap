// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCounter component FilmCounter component shows correctly with 1 movie 1`] = `
<span>
  1
   
  movie
   found
</span>
`;

exports[`FilmCounter component FilmCounter component shows correctly with many movies 1`] = `
<span>
  2
   
  movies
   found
</span>
`;
